firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManager.java

```diff
@@ -106,9 +108,19 @@ private static boolean hasAnalyticsTrigger(TriggeringCondition tc, String event)
     return tc.getEvent().getName().equals(event);
   }
 
-  private static boolean isActive(Clock clock, VanillaCampaignPayload vanillaPayload) {
-    long campaignStartTime = vanillaPayload.getCampaignStartTimeMillis();
-    long campaignEndTime = vanillaPayload.getCampaignEndTimeMillis();
+  private static boolean isActive(Clock clock, ThickContent content) {
+    long campaignStartTime;
+    long campaignEndTime;
+    if (content.getPayloadCase().equals(ThickContent.PayloadCase.VANILLA_PAYLOAD)) {
+      campaignStartTime = content.getVanillaPayload().getCampaignStartTimeMillis();
+      campaignEndTime = content.getVanillaPayload().getCampaignEndTimeMillis();
+    } else {
+      campaignStartTime =
+          content.getExperimentalPayload().getExperimentPayload().getExperimentStartTimeMillis();
+      campaignEndTime =
+          campaignStartTime
+              + content.getExperimentalPayload().getExperimentPayload().getTimeToLiveMillis();
```

Comment:
Have we validated that this actually mimics what the user sets in the console for the expt?

Reply:
Right now this does not, in the future we need to make a backend change to sync this. b/147681213